src/test/java/tfb/status/service/ShareResultsMailerTest.java

```diff
@@ -0,0 +1,51 @@
+package tfb.status.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import tfb.status.testlib.MailServer;
+import tfb.status.testlib.TestServicesInjector;
+
+/**
+ * Tests for {@link ShareResultsMailer}.
+ */
+@ExtendWith(TestServicesInjector.class)
+public final class ShareResultsMailerTest {
+  /**
+   * Verifies that the mailer sends an email with the expected title when the
+   * {@link ShareResultsMailer#onShareDirectoryFull(long, long)} method is
+   * invoked.
+   */
+  @Test
+  public void shareResultsMailer_shareDirectoryFullEmail(
+      MailServer mailServer,
+      ShareResultsMailer shareResultsMailer)
+      throws IOException, MessagingException {
+
+    // The mailer is a singleton and may have been called by other tests
+    // indirectly. We just need to make sure it sent an email as a result of
+    // this test.
+    List<MimeMessage> messages =getShareDirectoryFullEmails(mailServer);
+    int initialMessagesSize = messages.size();
+
+    shareResultsMailer.onShareDirectoryFull(1234, 5678);
+
+    messages = getShareDirectoryFullEmails(mailServer);
+
+    assertEquals(initialMessagesSize + 1, messages.size());
```

Comment:
This works now, but may cause an issue if you make the tests run in parallel. One of the other tests tries to share a results file after filling the share directory to capacity, which also sends this email. I'm curious what the right solution would be. One idea is that since the email contains the capacity and size in the body, I could add code to find a message with the specific numbers passed to `onShareDirectoryFull` above.

Reply:
Let's not solve this now, but please [annotate the test](https://github.com/TechEmpower/tfb-status/blob/f67ee2908b459adab597cf034503acd315bdf7c3/src/test/java/tfb/status/handler/HomeUpdatesHandlerTest.java#L24) so that it can't run in parallel and [add a `TODO`](https://github.com/TechEmpower/tfb-status/blob/f67ee2908b459adab597cf034503acd315bdf7c3/src/test/java/tfb/status/handler/HomeUpdatesHandlerTest.java#L41) near the part of the test that's unfriendly to parallelism.